crates/ruff_linter/resources/test/fixtures/flake8_comprehensions/C417.py

```diff
@@ -75,3 +75,11 @@ def func(arg1: int, arg2: int = 4):
 _ = t"{set(map(lambda x: x % 2 == 0, nums))}"
 _ = t"{dict(map(lambda v: (v, v**2), nums))}"
 
+
+# See https://github.com/astral-sh/ruff/issues/20198
+# No error: lambda contains `yield`, so map() should not be rewritten
+base = 0
+gens = map(lambda x: (yield base + x), [1, 2, 3])
+for gen in gens:
+    _ = [*gen]
+    base += 20
```

Comment:
nit: We might want to strip this down a bit like the other test cases:


```suggestion
map(lambda x: (yield x), [1, 2, 3])
```